Assignment 2/src/main/java/com/nuclei/assignment/service/menuoptions/MenuOptionsImpl.java

```diff
@@ -0,0 +1,187 @@
+package com.nuclei.assignment.service.menuoptions;
+
+import com.nuclei.assignment.constants.ExceptionsConstantsUtils;
+import com.nuclei.assignment.constants.StringConstantsUtils;
+import com.nuclei.assignment.constants.SuccessConstantsUtils;
+import com.nuclei.assignment.entity.UserEntity;
+import com.nuclei.assignment.enums.Courses;
+import com.nuclei.assignment.enums.SortingOrder;
+import com.nuclei.assignment.exception.CustomException;
+import com.nuclei.assignment.service.inputparser.InputParser;
+import com.nuclei.assignment.service.inputparser.InputParserImpl;
+import com.nuclei.assignment.service.useroperations.UserOperations;
+import com.nuclei.assignment.service.useroperations.UserOperationsImpl;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Scanner;
+import java.util.Set;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Menu Option implementation for different types of access.
+ */
+public class MenuOptionsImpl implements MenuOptions {
+  
+  /**
+   * The User operations.
+   */
+  private final UserOperations userOperations;
+  
+  /**
+   * The Input Parser.
+   */
+  private final InputParser inputParser;
+  
+  /**
+   * The Logger.
+   */
+  private final Log logger = LogFactory.getLog(MenuOptionsImpl.class);
+  
+  /**
+   * Menu Option Constructor.
+   */
+  public MenuOptionsImpl() {
+    userOperations = new UserOperationsImpl();
+    inputParser = new InputParserImpl();
+  }
+  
+  @Override
+  public void menuInterface() {
+    logger.info("New user session started");
```

Comment:
make it either debug, or remove it if it is not fulfiling any purpose.

Reply:
done